### PR TITLE
don't react to disposing view types if the workbench is shutting down

### LINKED ISSUE #176311: Untitled notebook will prompt to save on window reload or close
Steps to Reproduce:

1. close any open workspace
2. set "files.hotExit": "onExitAndWindowClose" to not be prompted about saving on close
3. Create a new notebook and add some code
4. reload the window
:bug: save prompt is shown, clicking 'save' doesn't do anything

### PR DESCRIPTION
**Pull request by amunger**
Fixes https://github.com/microsoft/vscode/issues/176311#issuecomment-1457572937

If the workbench is shutting down, we don't need to worry about closing down editors in reaction to losing a `viewtype`.

### CODE REVIEW

**Review by bpasero — CHANGES_REQUESTED** [review]
Why not just have the listener in an extra `IDisposable` that gets disposed `onWillShutdown`?

**Review by jrieken — APPROVED** [review]

File: src/vs/workbench/contrib/notebook/browser/notebook.contribution.ts [hunk 1]
```diff
@@ -570,14 +564,18 @@ class NotebookEditorManager implements IWorkbenchContribution {
...
+			for (const group of editorGroups.groups) {
+				const staleInputs = group.editors.filter(input => input instanceof NotebookEditorInput && input.viewType === e);
+				group.closeEditors(staleInputs);
 			}
-		}));
+		});
+
+		this._disposables.add(listener);
+		// don't react to view types disposing if the workbench is shutting down anyway
+		lifecycleService.onWillShutdown((e) => {
```
[the remark this answers is not in the document]
Comment (bpasero) [reply src/vs/workbench/contrib/notebook/browser/notebook.contribution.ts]: I think he pushed a change to use `Event.once` later.

**Review by bpasero — APPROVED** [review]
Awesome!

File: src/vs/workbench/contrib/notebook/browser/notebook.contribution.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (jrieken) [reply src/vs/workbench/contrib/notebook/browser/notebook.contribution.ts]: Oh, I dislike GH PR notifications very much.

**Pull request merged**